spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/ReactiveGridFsResource.java

```diff
@@ -33,71 +34,77 @@
 import org.springframework.util.Assert;
 
 import com.mongodb.client.gridfs.model.GridFSFile;
-import com.mongodb.reactivestreams.client.gridfs.GridFSDownloadPublisher;
+import com.mongodb.reactivestreams.client.gridfs.GridFSBucket;
 
 /**
  * Reactive {@link GridFSFile} based {@link Resource} implementation. Note that the {@link #getDownloadStream() content}
  * can be consumed only once.
  *
  * @author Mark Paluch
  * @author Christoph Strobl
+ * @author Mathieu Ouellet
  * @since 2.2
  */
 public class ReactiveGridFsResource implements GridFsObject<Object, Publisher<DataBuffer>> {
 
 	private final AtomicBoolean consumed = new AtomicBoolean(false);
 
 	private final @Nullable Object id;
+	private final @Nullable GridFSFile file;
 	private final Options options;
 	private final String filename;
-	private final @Nullable GridFSDownloadPublisher downloadPublisher;
+	private final Mono<GridFSBucket> bucketPublisher;
 	private final DataBufferFactory dataBufferFactory;
 
 	/**
 	 * Creates a new, absent {@link ReactiveGridFsResource}.
 	 *
 	 * @param filename filename of the absent resource.
-	 * @param downloadPublisher
+	 * @param bucketPublisher
 	 */
-	public ReactiveGridFsResource(String filename, @Nullable GridFSDownloadPublisher downloadPublisher) {
-		this(null, filename, Options.none(), downloadPublisher);
+	public ReactiveGridFsResource(String filename, Mono<GridFSBucket> bucketPublisher) {
```

Comment:
This change should not be necessary as `ReactiveGridFsResource` uses already deferred creation. See `ReactiveGridFsTemplate.getResource(…)`

Reply:
Fixed!